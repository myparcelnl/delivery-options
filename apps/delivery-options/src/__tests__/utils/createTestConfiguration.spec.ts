import {describe, it, expect} from 'vitest';
import {createTestConfiguration} from './createTestConfiguration';

describe('createTestConfiguration', () => {
  it('creates a default test configuration with unique prices for all options', () => {
    const result = createTestConfiguration();

    expect(result).toMatchSnapshot();
  });
});
