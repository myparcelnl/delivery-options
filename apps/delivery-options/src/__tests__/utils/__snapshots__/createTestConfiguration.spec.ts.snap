// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createTestConfiguration > creates a default test configuration with unique prices for all options 1`] = `
{
  "address": {
    "cc": "NL",
    "city": "Hoofddorp",
    "postalCode": "2132 JE",
    "street": "Antareslaan 31",
  },
  "config": {
    "allowDeliveryOptions": true,
    "allowEveningDelivery": true,
    "allowMondayDelivery": true,
    "allowMorningDelivery": true,
    "allowOnlyRecipient": true,
    "allowPickupLocations": true,
    "allowSameDayDelivery": true,
    "allowSaturdayDelivery": true,
    "allowSignature": true,
    "allowStandardDelivery": true,
    "carrierSettings": {
      "bpost": {
        "priceEveningDelivery": 2097168,
        "priceMondayDelivery": 2097184,
        "priceMorningDelivery": 2097160,
        "priceOnlyRecipient": 2097664,
        "pricePackageTypeDigitalStamp": 2097154,
        "pricePackageTypeMailbox": 2097153,
        "pricePackageTypePackageSmall": 2097156,
        "pricePickup": 2097408,
        "priceSameDayDelivery": 2097280,
        "priceSaturdayDelivery": 2097216,
        "priceSignature": 2097408,
        "priceStandardDelivery": 2097152,
      },
      "bpost:1": {
        "priceEveningDelivery": 2359312,
        "priceMondayDelivery": 2359328,
        "priceMorningDelivery": 2359304,
        "priceOnlyRecipient": 2359808,
        "pricePackageTypeDigitalStamp": 2359298,
        "pricePackageTypeMailbox": 2359297,
        "pricePackageTypePackageSmall": 2359300,
        "pricePickup": 2359552,
        "priceSameDayDelivery": 2359424,
        "priceSaturdayDelivery": 2359360,
        "priceSignature": 2359552,
        "priceStandardDelivery": 2359296,
      },
      "bpost:2": {
        "priceEveningDelivery": 2621456,
        "priceMondayDelivery": 2621472,
        "priceMorningDelivery": 2621448,
        "priceOnlyRecipient": 2621952,
        "pricePackageTypeDigitalStamp": 2621442,
        "pricePackageTypeMailbox": 2621441,
        "pricePackageTypePackageSmall": 2621444,
        "pricePickup": 2621696,
        "priceSameDayDelivery": 2621568,
        "priceSaturdayDelivery": 2621504,
        "priceSignature": 2621696,
        "priceStandardDelivery": 2621440,
      },
      "dhl": {
        "priceEveningDelivery": 33554448,
        "priceMondayDelivery": 33554464,
        "priceMorningDelivery": 33554440,
        "priceOnlyRecipient": 33554944,
        "pricePackageTypeDigitalStamp": 33554434,
        "pricePackageTypeMailbox": 33554433,
        "pricePackageTypePackageSmall": 33554436,
        "pricePickup": 33554688,
        "priceSameDayDelivery": 33554560,
        "priceSaturdayDelivery": 33554496,
        "priceSignature": 33554688,
        "priceStandardDelivery": 33554432,
      },
      "dhl:1": {
        "priceEveningDelivery": 33816592,
        "priceMondayDelivery": 33816608,
        "priceMorningDelivery": 33816584,
        "priceOnlyRecipient": 33817088,
        "pricePackageTypeDigitalStamp": 33816578,
        "pricePackageTypeMailbox": 33816577,
        "pricePackageTypePackageSmall": 33816580,
        "pricePickup": 33816832,
        "priceSameDayDelivery": 33816704,
        "priceSaturdayDelivery": 33816640,
        "priceSignature": 33816832,
        "priceStandardDelivery": 33816576,
      },
      "dhl:2": {
        "priceEveningDelivery": 34078736,
        "priceMondayDelivery": 34078752,
        "priceMorningDelivery": 34078728,
        "priceOnlyRecipient": 34079232,
        "pricePackageTypeDigitalStamp": 34078722,
        "pricePackageTypeMailbox": 34078721,
        "pricePackageTypePackageSmall": 34078724,
        "pricePickup": 34078976,
        "priceSameDayDelivery": 34078848,
        "priceSaturdayDelivery": 34078784,
        "priceSignature": 34078976,
        "priceStandardDelivery": 34078720,
      },
      "dhleuroplus": {
        "priceEveningDelivery": 1073741840,
        "priceMondayDelivery": 1073741856,
        "priceMorningDelivery": 1073741832,
        "priceOnlyRecipient": 1073742336,
        "pricePackageTypeDigitalStamp": 1073741826,
        "pricePackageTypeMailbox": 1073741825,
        "pricePackageTypePackageSmall": 1073741828,
        "pricePickup": 1073742080,
        "priceSameDayDelivery": 1073741952,
        "priceSaturdayDelivery": 1073741888,
        "priceSignature": 1073742080,
        "priceStandardDelivery": 1073741824,
      },
      "dhleuroplus:1": {
        "priceEveningDelivery": 1074003984,
        "priceMondayDelivery": 1074004000,
        "priceMorningDelivery": 1074003976,
        "priceOnlyRecipient": 1074004480,
        "pricePackageTypeDigitalStamp": 1074003970,
        "pricePackageTypeMailbox": 1074003969,
        "pricePackageTypePackageSmall": 1074003972,
        "pricePickup": 1074004224,
        "priceSameDayDelivery": 1074004096,
        "priceSaturdayDelivery": 1074004032,
        "priceSignature": 1074004224,
        "priceStandardDelivery": 1074003968,
      },
      "dhleuroplus:2": {
        "priceEveningDelivery": 1074266128,
        "priceMondayDelivery": 1074266144,
        "priceMorningDelivery": 1074266120,
        "priceOnlyRecipient": 1074266624,
        "pricePackageTypeDigitalStamp": 1074266114,
        "pricePackageTypeMailbox": 1074266113,
        "pricePackageTypePackageSmall": 1074266116,
        "pricePickup": 1074266368,
        "priceSameDayDelivery": 1074266240,
        "priceSaturdayDelivery": 1074266176,
        "priceSignature": 1074266368,
        "priceStandardDelivery": 1074266112,
      },
      "dhlforyou": {
        "priceEveningDelivery": 268435472,
        "priceMondayDelivery": 268435488,
        "priceMorningDelivery": 268435464,
        "priceOnlyRecipient": 268435968,
        "pricePackageTypeDigitalStamp": 268435458,
        "pricePackageTypeMailbox": 268435457,
        "pricePackageTypePackageSmall": 268435460,
        "pricePickup": 268435712,
        "priceSameDayDelivery": 268435584,
        "priceSaturdayDelivery": 268435520,
        "priceSignature": 268435712,
        "priceStandardDelivery": 268435456,
      },
      "dhlforyou:1": {
        "priceEveningDelivery": 268697616,
        "priceMondayDelivery": 268697632,
        "priceMorningDelivery": 268697608,
        "priceOnlyRecipient": 268698112,
        "pricePackageTypeDigitalStamp": 268697602,
        "pricePackageTypeMailbox": 268697601,
        "pricePackageTypePackageSmall": 268697604,
        "pricePickup": 268697856,
        "priceSameDayDelivery": 268697728,
        "priceSaturdayDelivery": 268697664,
        "priceSignature": 268697856,
        "priceStandardDelivery": 268697600,
      },
      "dhlforyou:2": {
        "priceEveningDelivery": 268959760,
        "priceMondayDelivery": 268959776,
        "priceMorningDelivery": 268959752,
        "priceOnlyRecipient": 268960256,
        "pricePackageTypeDigitalStamp": 268959746,
        "pricePackageTypeMailbox": 268959745,
        "pricePackageTypePackageSmall": 268959748,
        "pricePickup": 268960000,
        "priceSameDayDelivery": 268959872,
        "priceSaturdayDelivery": 268959808,
        "priceSignature": 268960000,
        "priceStandardDelivery": 268959744,
      },
      "dhlparcelconnect": {
        "priceEveningDelivery": 536870928,
        "priceMondayDelivery": 536870944,
        "priceMorningDelivery": 536870920,
        "priceOnlyRecipient": 536871424,
        "pricePackageTypeDigitalStamp": 536870914,
        "pricePackageTypeMailbox": 536870913,
        "pricePackageTypePackageSmall": 536870916,
        "pricePickup": 536871168,
        "priceSameDayDelivery": 536871040,
        "priceSaturdayDelivery": 536870976,
        "priceSignature": 536871168,
        "priceStandardDelivery": 536870912,
      },
      "dhlparcelconnect:1": {
        "priceEveningDelivery": 537133072,
        "priceMondayDelivery": 537133088,
        "priceMorningDelivery": 537133064,
        "priceOnlyRecipient": 537133568,
        "pricePackageTypeDigitalStamp": 537133058,
        "pricePackageTypeMailbox": 537133057,
        "pricePackageTypePackageSmall": 537133060,
        "pricePickup": 537133312,
        "priceSameDayDelivery": 537133184,
        "priceSaturdayDelivery": 537133120,
        "priceSignature": 537133312,
        "priceStandardDelivery": 537133056,
      },
      "dhlparcelconnect:2": {
        "priceEveningDelivery": 537395216,
        "priceMondayDelivery": 537395232,
        "priceMorningDelivery": 537395208,
        "priceOnlyRecipient": 537395712,
        "pricePackageTypeDigitalStamp": 537395202,
        "pricePackageTypeMailbox": 537395201,
        "pricePackageTypePackageSmall": 537395204,
        "pricePickup": 537395456,
        "priceSameDayDelivery": 537395328,
        "priceSaturdayDelivery": 537395264,
        "priceSignature": 537395456,
        "priceStandardDelivery": 537395200,
      },
      "dpd": {
        "priceEveningDelivery": 8388624,
        "priceMondayDelivery": 8388640,
        "priceMorningDelivery": 8388616,
        "priceOnlyRecipient": 8389120,
        "pricePackageTypeDigitalStamp": 8388610,
        "pricePackageTypeMailbox": 8388609,
        "pricePackageTypePackageSmall": 8388612,
        "pricePickup": 8388864,
        "priceSameDayDelivery": 8388736,
        "priceSaturdayDelivery": 8388672,
        "priceSignature": 8388864,
        "priceStandardDelivery": 8388608,
      },
      "dpd:1": {
        "priceEveningDelivery": 8650768,
        "priceMondayDelivery": 8650784,
        "priceMorningDelivery": 8650760,
        "priceOnlyRecipient": 8651264,
        "pricePackageTypeDigitalStamp": 8650754,
        "pricePackageTypeMailbox": 8650753,
        "pricePackageTypePackageSmall": 8650756,
        "pricePickup": 8651008,
        "priceSameDayDelivery": 8650880,
        "priceSaturdayDelivery": 8650816,
        "priceSignature": 8651008,
        "priceStandardDelivery": 8650752,
      },
      "dpd:2": {
        "priceEveningDelivery": 8912912,
        "priceMondayDelivery": 8912928,
        "priceMorningDelivery": 8912904,
        "priceOnlyRecipient": 8913408,
        "pricePackageTypeDigitalStamp": 8912898,
        "pricePackageTypeMailbox": 8912897,
        "pricePackageTypePackageSmall": 8912900,
        "pricePickup": 8913152,
        "priceSameDayDelivery": 8913024,
        "priceSaturdayDelivery": 8912960,
        "priceSignature": 8913152,
        "priceStandardDelivery": 8912896,
      },
      "postnl": {
        "priceEveningDelivery": 1048592,
        "priceMondayDelivery": 1048608,
        "priceMorningDelivery": 1048584,
        "priceOnlyRecipient": 1049088,
        "pricePackageTypeDigitalStamp": 1048578,
        "pricePackageTypeMailbox": 1048577,
        "pricePackageTypePackageSmall": 1048580,
        "pricePickup": 1048832,
        "priceSameDayDelivery": 1048704,
        "priceSaturdayDelivery": 1048640,
        "priceSignature": 1048832,
        "priceStandardDelivery": 1048576,
      },
      "postnl:1": {
        "priceEveningDelivery": 1310736,
        "priceMondayDelivery": 1310752,
        "priceMorningDelivery": 1310728,
        "priceOnlyRecipient": 1311232,
        "pricePackageTypeDigitalStamp": 1310722,
        "pricePackageTypeMailbox": 1310721,
        "pricePackageTypePackageSmall": 1310724,
        "pricePickup": 1310976,
        "priceSameDayDelivery": 1310848,
        "priceSaturdayDelivery": 1310784,
        "priceSignature": 1310976,
        "priceStandardDelivery": 1310720,
      },
      "postnl:2": {
        "priceEveningDelivery": 1572880,
        "priceMondayDelivery": 1572896,
        "priceMorningDelivery": 1572872,
        "priceOnlyRecipient": 1573376,
        "pricePackageTypeDigitalStamp": 1572866,
        "pricePackageTypeMailbox": 1572865,
        "pricePackageTypePackageSmall": 1572868,
        "pricePickup": 1573120,
        "priceSameDayDelivery": 1572992,
        "priceSaturdayDelivery": 1572928,
        "priceSignature": 1573120,
        "priceStandardDelivery": 1572864,
      },
      "ups": {
        "priceEveningDelivery": 134217744,
        "priceMondayDelivery": 134217760,
        "priceMorningDelivery": 134217736,
        "priceOnlyRecipient": 134218240,
        "pricePackageTypeDigitalStamp": 134217730,
        "pricePackageTypeMailbox": 134217729,
        "pricePackageTypePackageSmall": 134217732,
        "pricePickup": 134217984,
        "priceSameDayDelivery": 134217856,
        "priceSaturdayDelivery": 134217792,
        "priceSignature": 134217984,
        "priceStandardDelivery": 134217728,
      },
      "ups:1": {
        "priceEveningDelivery": 134479888,
        "priceMondayDelivery": 134479904,
        "priceMorningDelivery": 134479880,
        "priceOnlyRecipient": 134480384,
        "pricePackageTypeDigitalStamp": 134479874,
        "pricePackageTypeMailbox": 134479873,
        "pricePackageTypePackageSmall": 134479876,
        "pricePickup": 134480128,
        "priceSameDayDelivery": 134480000,
        "priceSaturdayDelivery": 134479936,
        "priceSignature": 134480128,
        "priceStandardDelivery": 134479872,
      },
      "ups:2": {
        "priceEveningDelivery": 134742032,
        "priceMondayDelivery": 134742048,
        "priceMorningDelivery": 134742024,
        "priceOnlyRecipient": 134742528,
        "pricePackageTypeDigitalStamp": 134742018,
        "pricePackageTypeMailbox": 134742017,
        "pricePackageTypePackageSmall": 134742020,
        "pricePickup": 134742272,
        "priceSameDayDelivery": 134742144,
        "priceSaturdayDelivery": 134742080,
        "priceSignature": 134742272,
        "priceStandardDelivery": 134742016,
      },
    },
    "priceEveningDelivery": 16,
    "priceMondayDelivery": 32,
    "priceMorningDelivery": 8,
    "priceOnlyRecipient": 512,
    "pricePackageTypeDigitalStamp": 2,
    "pricePackageTypeMailbox": 1,
    "pricePackageTypePackageSmall": 4,
    "pricePickup": 256,
    "priceSameDayDelivery": 128,
    "priceSaturdayDelivery": 64,
    "priceSignature": 256,
    "priceStandardDelivery": 0,
    "showDeliveryDate": true,
  },
  "strings": {
    "addressNotFound": "Adresgegevens niet ingevuld",
    "city": "Plaats",
    "closed": "Gesloten",
    "deliveryTitle": "Thuis of op het werk bezorgen",
    "discount": "Korting",
    "error3212": "{} is verplicht.",
    "error3224": "",
    "error3501": "Dit adres is niet gevonden.",
    "error3505": "Postcode is niet geldig voor het gekozen land.",
    "error3506": "",
    "error3728": "",
    "from": "Vanaf",
    "headerDeliveryOptions": "",
    "number": "Huisnummer",
    "onlyRecipientTitle": "Alleen ontvanger",
    "openingHours": "Openingstijden",
    "options": "Opties",
    "packageTypeDigitalStamp": "Digitale postzegel",
    "packageTypeMailbox": "Brievenbuspakje",
    "pickUp": "Afhalen",
    "pickUpFrom": "Afhalen vanaf",
    "pickupLocationsListButton": "Lijst",
    "pickupLocationsMapButton": "Kaart",
    "pickupTitle": "Afhalen op locatie",
    "postalCode": "Postcode",
    "showMoreHours": "Toon meer openingstijden",
    "showMoreLocations": "Toon meer locaties",
    "signatureTitle": "Handtekening voor ontvangst",
    "street": "Straat",
  },
}
`;
